apps/service_discovery_grpc/src/service_discovery_grpc_app.erl

```diff
@@ -9,8 +9,14 @@
 
 -export([start/2, stop/1]).
 
+-define(DEFAULT_GRPC_SERVER, #{grpc_opts => #{service_protos => [sdg_discovery_pb],
+                                              services => #{'sd.DiscoveryService' => sdg_service}},
+                               listen_opts => #{port => 8081,
+                                                ip => {0,0,0,0}}}).
+
 start(_StartType, _StartArgs) ->
-    service_discovery_grpc_sup:start_link().
+    ServerOpts = application:get_env(service_discovery_grpc, server, ?DEFAULT_GRPC_SERVER),
```

Comment:
risk here: if you put the lookup in the supervisor's init itself, you'll allow:

1. the ability to derive the opts in the init call and not have to carry it in its parent's state (root sup)
2. the ability to restart the server and reload its config after a crash without restarting the whole app.

for now this makes the config static for the entire app, though it gives a more functional approach should someone want to start their own.

Reply:
Yea, my thought was you want to not get a different config on a crash and restart.

But you do want to get a fresh config when you restart the app.